syntax = "proto3";

package example;

// Define the request and response message types
message HelloRequest {
  string name = 1;
}

message FactoryRequest {
  string user_query = 1;
  string token = 2;
}

message HelloResponse {
  string message = 1;
}

message FactoryResponse {
  string response_message = 1;
}

// Define the service
service SmartFactory {
  rpc SayHello (HelloRequest) returns (HelloResponse);
  rpc Factory (FactoryRequest) returns (FactoryResponse);
}
